src/Encoder.h

```diff
@@ -10,33 +17,59 @@ namespace cppkin
 {
     class Span;
     class SimpleAnnotation;
-    class EncoderContext;
 
-    template<EncodingTypes::Enumeration >
-    class Encoder
+    class Encoder {
+    public:
+        virtual ~Encoder() {}
+        virtual std::string ToString(const Span&) const = 0;
+        virtual std::string ToString(const std::vector<EncoderContext::ContextElement>&) const = 0;
+        virtual std::string ToString(const std::vector<Span*>&) const = 0;
```

Comment:
Why do we need this method? who is going to address the Encoder by not using an EncoderContext?